hoomd/extern/dfftlib/src/dfft_cuda.cc

```diff
@@ -677,17 +660,11 @@ void dfft_cuda_redistribute_cyclic_to_block_1d(int *dim,
         // copy back received data
         hipMemcpy(d_work,h_stage_out, sizeof(cuda_cpx_t)*npackets*size,hipMemcpyDefault);
         if (check_err) CHECK_CUDA();
-        #endif
         }
     else
         {
         /* perform communication */
         MPI_Barrier(comm);
-        #ifdef ENABLE_MPI_CUDA
-        MPI_Alltoallv(d_work,dfft_nsend, dfft_offset_send, MPI_BYTE,
-                      d_scratch, dfft_nrecv, dfft_offset_recv, MPI_BYTE,
-                      comm);
-        #else
```

Comment:
The `#endif` corresponding to this block (on line 679) also needs to be removed.